Reset state instead of creating duplicate modals

mattheu opened the pull request:
I have picked up the work started by @rahulsprajapati over here: https://github.com/wp-shortcake/shortcake/pull/680/files. This is a tricky problem, but one that we should definitely try and fix.

I tested his code, but I was still seeing some duplicate markup - as you said when clicking the standard add media button instead.

This fix takes a slightly different approach. Instead it tries to always use the same media modal. If it doesn't exist, it creates it. If it does, it updates the state of the existing one. Additionally I had to do a bit of work to reset the state too, to ensure that when a user clicks a button again, they do not see the old state. I guess we avoided this issue before because it just created a new instance of the modal!

I'd appreciate your thoughts on this.

mattheu committed:
Reset state instead of creating duplicate modals

mattheu committed:
Set state back to insert on close/submit

mattheu committed:
Update button text

mattheu committed:
Compile

mattheu committed:
Merge remote-tracking branch 'origin/master' into fix-duplicate-markup

mattheu committed:
Update menu text

goldenapples commented on js/src/controllers/media-controller.js:
```diff
@@ -47,6 +47,49 @@ var MediaController = wp.media.controller.State.extend({
+
+	},
+
+	toggleSidebar: function( show ) {
```
Very minor nit - could this variable be named `hide` rather than `show`, since it determines whether or not the sidebar will be hidden.

goldenapples committed:
Reset the currentShortcode value and the menu title on close

We're correctly setting the state of the WP media modal back to
'insert', as used by WP core, when a user closes, submits, or dismisses
the shortcode UI modal. However the currentShortcode state wasn't being
reset, and the modal title wasn't being reset, so the next time the user
opened the media modal, they'd see "{Shortcode} Options" as the menu
item title, rather than "Insert Post Element" as it should be.

This change deletes and resets these props on closing the shortcode ui
frame. It could certainly stand to be refactored, but I didn't see the
best way of doing that offhand.

goldenapples committed:
Merge pull request #705 from wp-shortcake/reset-currentshortcode-and-menu-title-on-close

Reset the currentShortcode value and the menu title on close
